### ISSUE TITLE
Assertion error when NaN's in database (HITEMP)

### ISSUE THREAD

**Issue by dcmvdbekerom** [radis/radis#111]
## Expected Behavior
When synthesizing CO2 spectra, the HITEMP database has some unassaigned lines which have NaN values for their quantum numbers. When calculating a spectrum I would expect these lines to be ignored.

## Current Behavior
Instead of ignoring, an assertion is made that there are no NaN values. When the assertion fails, the following error is thrown:

```
Traceback (most recent call last):
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\base.py", line 305, in assert_no_nan
    assert np.isnan(df[column]).sum() == 0
AssertionError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:/Users/dcmvd/Documents/Hybrid plasma/CO2 Emission/co2_HITEMP_min_ex.py", line 17, in <module>
    path_length=20,# cm
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\calc.py", line 411, in calc_spectrum
    name=name,
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\factory.py", line 960, in non_eq_spectrum
    self._check_noneq_parameters(vib_distribution, singleTvibmode)
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\base.py", line 1551, in _check_noneq_parameters
    calc_Evib_harmonic_anharmonic=calc_Evib_harmonic_anharmonic,
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\base.py", line 1662, in _calc_noneq_parameters
    df, calc_Evib_harmonic_anharmonic=calc_Evib_harmonic_anharmonic
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\base.py", line 428, in _add_Evib123Erot
    return self._add_Evib123Erot_RADIS_cls5(df)
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\base.py", line 1236, in _add_Evib123Erot_RADIS_cls5
    self.assert_no_nan(df, "Evib1u")
  File "C:\Users\dcmvd\AppData\Local\Programs\Python\Python37\lib\site-packages\radis\lbl\base.py", line 311, in assert_no_nan
    index, get_print_full(df.loc[index])
AssertionError: Evib1u=NaN in line database at index 225 corresponding to Line:
 id            2.000000e+00
iso           1.000000e+00
wav           2.195028e+03
int           6.838000e-45
A             1.870000e+02
airbrd        6.840000e-02
selbrd        7.100000e-02
El            1.303464e+04
Tdpair        7.200000e-01
Pshft         0.000000e+00
branch        1.000000e+00
jl            6.200000e+01
v1u                    NaN
v2u                    NaN
l2u                    NaN
v3u                    NaN
ru                     NaN
v1l           4.000000e+00
v2l           2.000000e+00
l2l           2.000000e+00
v3l           2.000000e+00
rl            1.000000e+00
Ia            9.842040e-01
molar_mass    4.398983e+01
ju            6.300000e+01
Eu            1.522967e+04
Evib1l        5.300560e+03
Evib2l        1.335700e+03
Evib3l        4.673380e+03
Evib1u                 NaN
Evib2u                 NaN
Evib3u                 NaN
Evibu                  NaN
Evibl         1.130964e+04
Erotu                  NaN
Erotl         1.724999e+03
Name: 225, dtype: float64
```

## Possible Solution
Instead of asserting there are no NaN's, perhaps it would be better to ignore these lines and remove them from the database. The rationale is that these lines have to be quite "exotic" to not to have any quantum number assigned and therefore (hopefully) uncommon and/or low in intensity.

Removing lines with NaN's could be done with df.dropna()

A warning that NaN lines were removed would be courteous. 

## Steps to Reproduce (for bugs)
Minimal code to reproduce error:
```
from radis import calc_spectrum

T     =  500 #K
T12   =  600 #K
T3    = 1600 #K

s = calc_spectrum(2200,     # cm-1
                  2400,     # cm-1
                  databank = 'HITEMP',
                  molecule='CO2',
                  isotope='1,2,3',
                  pressure=80*1e-3,# bar
                  Tgas = T,
                  Trot = T,
                  Tvib = (T12,T12,T3),
                  mole_fraction=0.001,
                  path_length=20,# cm
                  )

s.apply_slit(0.400, 'cm-1',shape='gaussian')
s.plot()
```

My .radis file:

```
[HITEMP]                      
info = HITEMP databank 
path =                   
       C:\HITEMP\02_2000-2125_HITEMP2010.par
       C:\HITEMP\02_2125-2250_HITEMP2010.par
       C:\HITEMP\02_2250-2500_HITEMP2010.par
format = hitran                                     
levelsfmt: radis                             
parfuncfmt: hapi   
```

CO2 HITEMP database [here](https://hitran.org/hitemp/data/HITEMP-2010/CO2_line_list/).

## Context
Because of this error I am unable to use HITEMP database for synthesizing non-equilibrium CO2-spectra.

**Comment by erwanp** [comment]
Hello @dcmvdbekerom 

So first thing:

I ran the same code on my computer and could calculate the spectra correctly. But I'm using the original [CDSD-HITEMP](ftp://ftp.iao.ru/pub/CDSD-HITEMP/) version of the Tomsk's group, which was included in the HITEMP-2010 database for CO2. These databases are supposed to be the same, but the format is slightly different and it might be possible that they lost some of the quantum numbers in the process. 

In #47 we had a similar problem, with actually resulted from an error in the HITRAN database labelling. The error was solved by ignoring the problematic lines, in the user script. 

We could discard the error and ignore the lines directly in the code itself. However it sounds dangerous as we don't want to make a decision that could change the physics without the user knowing it. 

I'd rather solve it on the user side, as in #47. But that required using SpectrumFactory instead of calc_spectrum, and editing the Line database manually (if you want to do fitting, you'll need SpectrumFactory anyway not to have to load the line database multiple times). 
There is a [discard_lines_with_na_levels](https://github.com/radis/radis/blob/develop/radis/test/utils.py#L318) function used in the test cases that does a similar thing. 

Would give something along the lines of (not tested): 
````
sf = SpectrumFactory(2200,     # cm-1
                  2400,     # cm-1
                  databank = 
                  molecule='CO2',
                  isotope='1,2,3',
                  pressure=80*1e-3,# bar
                  Tgas = T,
                  Trot = T,
                  Tvib = (T12,T12,T3),
                  mole_fraction=0.001,
                  path_length=20,# cm)
sf.load_databank('HITEMP')

# >>> HACK 
# Try first calculation: should crash while fetching Energies. We catch that error:
try:
    sf.non_eq_spectrum(Trot=T, Tgas=T, Tvib=(T12,T12,T3))
except AssertionError:
    sf.df0.dropna(inplace=True)

# Re-run calculation (should work!)
sf.non_eq_spectrum(Trot=T, Tgas=T, Tvib=(T12,T12,T3))
```

Notes:

- @CorentinGrimaldi had the same error, with the HITEMP database too : how did you fix it Corentin??

- If you're interested in multi-temperature fitting, you may want to have a look at the [radis-examples project](https://github.com/radis/radis-examples)

- Also, CO2 noneq model still uses [Bart's constants](https://radis.readthedocs.io/en/latest/references/references.html#co2), which are best used with a Treanor distribution.

**Comment by dcmvdbekerom** [comment]
Hi @erwanp,

There is no link to the CDSD-HITEMP download. Could you add it? I think this will probably solve my immediate problem.

I'll have a look at the solution in #47 as well, I have no problem solving it on the user side.

**Comment by erwanp** [comment]
@BlehMaks wasn't this your error ? I don't know why we couldn't find it initially ...

**Comment by erwanp** [comment]
Fixed with a nicer error message by #226 , suggesting the user what to do to manually edit the database  (better than RADIS touching the database itself)
